### ISSUE TITLE
[BUG] Vysor does not disconnect from phone when closing the Vysor window

### ISSUE THREAD

**Issue by clubtech** [koush/vysor.io#870]
This worked fine until a few versions ago and is happening when connecting wirelessly (I have not tested wired).
I tested this with a Windows 10 laptop and a MacBook Pro 16''.
When I launch Vysor and use it for more than a few minutes, then close the Vysor mirroring window, the mirroring session does not actually disconnect from the phone until a few minutes later (I assume that's when a timeout on the Android side is happening).
I can see this because the Vysor international keyboard does not disappear right away like it used to until a few minutes have passed from the disconnect time.

To make the session close immediately I need to close the Vysor mirroring windows AND the Vysor main window from the taskbar. Then the keyboard disappears from the phone right away and the sessions closes.

Bug log: https://gist.github.com/VysorLogs/8ec3dbd965138438ce92da282a787730 (without my email and android logcat, let me know if you need that).

I ruled out this is a Wifi issue as I tried to use Vysor wirelessly on another network today and the same thing was happening.

I did notice that if a start a mirroring sessions and immediately close the Vysor window (within a few seconds), the session does close properly without me needing to also close the main Vysor window from the taskbar but if you keep the session open for a few minutes or more it does not close right away when the mirroring window is closed.

This worked 100% fine up until a few versions ago.

**Comment by siam-backend-developer** [comment]
hello koush 
please solve this problem ctrl+v it's not working in 6.0 android

**Comment by clubtech** [comment]
@koush any chance you can take a look at this?

**Comment by koush** [comment]
@clubtech you were responding to another issue and that was extremely confusing, because your logs were unrelated to what I was debugging.

**Comment by koush** [comment]
@clubtech I deleted your comments in the other issue.

**Comment by koush** [comment]
Please verify if this issue happens while wired.

**Comment by clubtech** [comment]
My sincere apologies! I was watching the other ticket and somehow thought you replied to mine :/ 
Yes I can confirm this is also happening over wired connection. The thing is, If I close the Vysor mirroring window 1-2 minutes after I start mirroring, it does close the session properly on the phone. The international keyboard goes away and the screen is no longer blank (I have it configured to have the phone screen blank).
If however the mirroring session is longer (say 5-10 minutes) and then close the Vysor mirroring window, the session does not close on the phone until I either disconnect the cable or close the main Vysor window in the taskbar.

This happens both over USB or wireless.

**Comment by clubtech** [comment]
Is there any way to troubleshoot this issue? It seems to be happening both via wireless and USB connection.

**Comment by koush** [comment]
I have a rough idea what is going on, the TCP connection isn't timing out, and I need to send some keepalives to verify it is alive, and shut down if the other side is gone.
Can you tell me if this happens with the phone's screen on? Basically, disable the setting, and before disconnecting (after waiting the appropriate number of minutes), ensure the screen is on/wake. Then disconnect.

**Comment by clubtech** [comment]
Yes this is happening with the screen on as well. Oddly this started happening only a few Vysor versions ago, so not sure what has changed.
When I close the main Vysor window (not the mirroring window), the sessions always fully disconnects.
If I click the "Force restart" adb option in the main window, it will also immediately disconnect the session so the phone server does see these types of loss of connection.

It's only happening if you close the Vysor mirroring window after the session was open for a few minutes.

**Comment by koush** [comment]
Which phone is this?

**Comment by clubtech** [comment]
Galaxy s21 Ultra
